apps/central-scan/backend/src/fujitsu_scanner.ts

```diff
@@ -93,27 +93,32 @@ export class FujitsuScanner implements BatchScanner {
     );
   }
 
-  async isImprinterAttached(): Promise<boolean> {
+  isImprinterAttached(): Promise<boolean> {
     return new Promise((resolve) => {
       const process = streamExecFile('scanimage', [
-        '-d',
+        '--device-name',
         'fujitsu',
         '--endorser=yes',
         '--format=jpeg',
         '--dont-scan',
       ]);
 
       assert(process.stderr);
+      let stderr = '';
       process.stderr.on('data', (data: string) => {
-        // If there is no imprinter attached a message will be sent to stderr
-        // that the endorser parameter is readonly.
-        if (data.includes(EXPECTED_IMPRINTER_UNATTACHED_ERROR)) {
-          resolve(false);
```

Comment:
Here is the problematic early `resolve` call. This PR changes it so we always resolve in the `close` event handler.

In addition to happening too early, the way this is set up also means that `resolve` gets called twice. This is actually allowed by the JavaScript spec and the second call will just be ignored, but it is confusing to read.